packages/ai-chat/src/browser/frontend-chat-service.ts

```diff
@@ -0,0 +1,66 @@
+// *****************************************************************************
+// Copyright (C) 2024 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { ChatAgent, ChatServiceImpl, ParsedChatRequest } from '../common';
+import { PreferenceService } from '@theia/core/lib/browser';
+import { DEFAULT_CHAT_AGENT_PREF } from './ai-chat-preferences';
+
+@injectable()
+export class FrontendChatServiceImpl extends ChatServiceImpl {
```

Comment:
The default implementation can run in both frontend and backend.

This is the customized frontend variant which additionally is able to check the preferences (as they are scoped per frontend) for user customizations.

Whenever possible we tried to go for a base implementation in "common", so that the core services could be used in backend and frontend. The default use case will always be the frontend (especially in Theia IDE), however there are specialized Theia based applications out there with a strong notion of headless support and/or with a backend focus which is why we even have backend-only plugin support now. These tools then have the option to use the ChatService then within the backend.

Reply:
I don't see the ChatService being used anywhere in the back end, but still an instance is bound in `agent-backend-module.ts`. If we want to have an example, we should add one in the "example" folder, not bind by default.

Reply:
The `FrontendChatService` is still bound, correct. However I removed the backend binding by deleting `packages/ai-chat/src/node/agent-backend-module.ts` in commit https://github.com/eclipse-theia/theia/pull/14048/commits/4d96c19258c9a4196a6e54d42ab350f2cf2c5a54